searchcore/src/vespa/searchcore/proton/documentmetastore/lid_allocator.cpp

```diff
@@ -221,10 +191,10 @@ LidAllocator::constructFreeList(DocId lidLimit)
 
 namespace {
 
-class BlackListBlueprint : public SimpleLeafBlueprint
+class WhiteListBlueprint : public SimpleLeafBlueprint
 {
 private:
-    AttributeVector::SearchContext::UP _searchCtx;
+    const search::GrowableBitVector &_activeLids;
```

Comment:
how is this protected against concurrent modification and/or deletion?

Reply:
_activeLids is owned by document meta store which is kept alive until DocumentDB is destructed (which happens after all active queries for that document type are drained).